packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const baseClass = 'search-bar';
+const noop = () => {};
+
+class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.onChangeDebounced = debounce(this.props.debounceInMs, value => {
+      this.props.onChange(value);
+    });
+
+    this.state = {
+      searchTerm: '',
+      isInCompactMode: this.props.compact
+    };
+  }
+
+  handleChange = e => {
+    const { value } = e.target;
+    const { debounceInMs, onChange } = this.props;
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.onChangeDebounced(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
+    const { onSubmit } = this.props;
+
+    if (onSubmit) {
+      if (key.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    this.setState({
+      searchTerm: ''
+    });
+
+    this.inputRef.current.focus();
+  };
+
+  handleKeyDown = key => {
+    if (key.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    this.setState(state => ({
+      isInCompactMode: !state.isInCompactMode
+    }));
+  };
```

Comment:
awesome, but let's make it easier for a user and do `this.inputRef.current.focus();` after `setState` if it was openned:
```suggestion
  toggleCompactMode = () => {
    this.setState(prevState => ({
      isInCompactMode: !state.isInCompactMode
    }), () => {
      if (!this.state.isInCompactMode && this.inputRef.current) {
        this.inputRef.current.focus();
      }
    });
  };
```